### PR TITLE
Add TS documentation for `attrs` method

### PR DESCRIPTION
**Pull request by davilima6**
Kudos to:
- @ajeetshah: https://stackoverflow.com/a/61980137/462849
- @fingerartur: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/28597#issuecomment-781273822
- @jtbandes: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/28597#issuecomment-791812675

### CODE REVIEW

**Comment by davilima6** [comment]
Hi, @probablyup! Would you know who could review this one?

**Comment by probablyup** [comment]
@davilima6 @Igorbek would be a good choice I think, he does a ton of work with TS

File: sections/api/typescript.md [hunk 1]
```diff
@@ -178,6 +178,49 @@ This is the most complex example where we have specific properties for the styli
 the rest of the required properties by the **Header** through. That means when you use **Title** it will have
 the combined typing of both the styled requirements and the actual component requirements.
 
+### Using `attrs` method
+
+The `attrs` method accepts an Interpolation object, which can be either a string that will be merged into the rest of the component's props, or a function that receives the styled component's `props` as the first and only argument.
```
Comment (Igorbek) [inline sections/api/typescript.md:183]: The description does not seem to be accurate to me. The `attrs` method does not interpolate (using backtick or a function). It passes set of predefined prop values down to the decorated component. So the examples seem right but not the description.

I'd suggest to describe it similarly to the website:

```suggestion
This is a chainable method that attaches some props to a styled component. It accepts an object with partial props that will be merged into the rest of the component's props. Alternatively, a function can be passed that takes interior props and returns an object to merge.
```

File: sections/api/typescript.md [hunk 2]
```diff
@@ -178,6 +178,49 @@ This is the most complex example where we have specific properties for the styli
...
+
+```jsx
+import styled from 'styled-components';
+import Header from './Header';
+
+interface TitleProps {
+  readonly isActive: boolean;
+};
+
+const Title = styled(Header).attrs<TitleProps>({ isActive: true })<TitleProps>`
```
Comment (Igorbek) [inline sections/api/typescript.md:195]: not sure if generic props need to be specified explicitly, it should have been a partial object of the wrapped component/tag. I'd suggest using examples from the website or similar https://styled-components.com/docs/api#attrs

File: sections/api/typescript.md [hunk 3]
```diff
@@ -178,6 +178,49 @@ This is the most complex example where we have specific properties for the styli
 the rest of the required properties by the **Header** through. That means when you use **Title** it will have
 the combined typing of both the styled requirements and the actual component requirements.
 
+### Using `attrs` method
+
+This is a chainable method that attaches some props to a styled component. It accepts an object with partial props that will be merged into the rest of the component's props. Alternatively, a function can be passed that takes interior props and returns an object to merge.
+
+For the string case, make sure to type it "twice", like this:
+
+```jsx
```
Comment (probablyup) [inline sections/api/typescript.md:187]: This should be tsx

**Review by probablyup — APPROVED** [review]

**Comment by probablyup** [comment]
@davilima6 sorry for the long wait, if you can get this PR rebased I'll merge it

**Comment by tonicprism** [comment]
This PR was merged with main?

**Pull request closed without merging**